dds/DCPS/XTypes/TypeObject.h

```diff
@@ -322,6 +348,18 @@ namespace XTypes {
       , bound(a_bound)
       , element_identifier(a_element_identifier)
     {}
+
+    bool operator<(const PlainSequenceSElemDefn& other) const
+    {
+      if (header < other.header) return true;
+      if (other.header < header) return false;
+      if (bound < other.bound) return true;
+      if (other.bound < bound) return false;
+      if (element_identifier < other.element_identifier) return true;
+      if (other.element_identifier < element_identifier) return false;
+      return false;
```

Comment:
```suggestion
      if (header < other.header ||
          (header == other.header && bound < other.bound) ||
          (header == other.header && bound == other.bound && element_identifier < other.element_identifier)) {
        return true;
      }
      return false;
```

Reply:
I suggest similar changes to `operator<` of `PlainSequenceLElemDefn`, `PlainArraySElemDefn`, `PlainArrayLElemDefn`, `PlainMapSTypeDefn`, `PlainMapLTypeDefn`

Reply:
Return the if condition?

Reply:
It returns true or false for `operator<`.

Reply:
I meant this:

```
return header < other.header || (header == other.header && bound < other.bound) ||
  (header == other.header && bound == other.bound && element_identifier < other.element_identifier);
```

Reply:
Yea, that works.

Reply:
I based current implementation based on suggestion from @mitza-oci . My understanding is that it is a more 'standard' structure comparison algorithm. My vote would be to keep it, unless it gives wrong result.